test/e2e/e2eutil/check_util.go

```diff
@@ -0,0 +1,32 @@
+package e2eutil
+
+import (
+	"testing"
+
+	"github.com/operator-framework/operator-sdk/pkg/util/retryutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func DeploymentReplicaCheck(t *testing.T, kubeclient *kubernetes.Clientset, namespace, name string, replicas, retries int) error {
+	err := retryutil.Retry(retryInterval, retries, func() (done bool, err error) {
+		deployment, err := kubeclient.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{IncludeUninitialized: true})
+		if err != nil {
```

Comment:
We should only retry if the Deployment was not found(i.e not yet created). For any other error we should just return the error:
```Go
import (
    apierrors "k8s.io/apimachinery/pkg/api/errors"
)
...

if err != nil && apierrors.IsNotFound(err) {
    ....
    return false, nil
} else {
    return false, err
}
```